File: leetcode/1_Two Sum.py
from collections import defaultdict
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = defaultdict(int)
        for idx, val in enumerate(nums):
            dic[val]=idx
        for idx, val in enumerate(nums):
            if ((target-val) in dic) and dic[target-val]!=idx:
                return [idx,dic[target-val]]
                
            

File: leetcode/1_Two Sum.cpp
// 요구사항: 합쳐서 target이 되는 두 수의 index를 반환해라 
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashMap; 
        
        for (int idx = 0; idx < nums.size(); idx++) {
            int diff = target - nums[idx]; 
            if ( hashMap.find(diff) != hashMap.end() ) {
                return {idx, hashMap[diff]};
            } else {
                hashMap[nums[idx]] = idx;
            }
        }
        return {};
    }
};